op.c

```diff
@@ -8022,18 +8022,25 @@ Perl_utilize(pTHX_ int aver, I32 floor, OP *version, OP *idop, OP *arg)
             /* use VERSION while another use VERSION is in scope
              * This should provoke at least a warning, if not an outright error
              */
-            if (shortver >= SHORTVER(5, 39))
+            if (PL_prevailing_version < SHORTVER(5, 10)) {
```

Comment:
Why not `... || shortver < SHORTVER(5, 10)`, too? It seems a bit weird that `use v5.8; use v5.12;` is accepted because `use v5.8` has no side effects, but `use v5.12; use v5.8;` is an error.

Reply:
Perl starts with implicit 5.8 semantics. So changing from 5.8 to 5.8 does nothing. `use` doesn't just add behaviors, but changes all of the semantics to match the requested version. Based on the existing behavior of use, the `use v5.8` in `use v5.12; use v5.8;` would have side effects, and disabling the 5.12 features and turning off strict. This behavior is weird enough (and I don't believe was ever implemented) that it has already been deprecated for 2 releases, and was already intended to be fatal in 5.40 even before we started looking at other changes.

Reply:
Oh, looks like I had misunderstood `use VERSION` semantics entirely. :-)
Thanks for setting me straight.